packages/model-viewer/src/test/features/scene-graph/model-spec.ts

```diff
@@ -74,5 +79,58 @@ suite('scene-graph/model', () => {
 
       expect(collectedMaterials.size).to.be.equal(materials.size);
     });
+
+    suite('Model Variants', () => {
+      test('Switch variant and lazy load', async () => {
+        const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+        const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+
+        expect(model[$materials][4][$correlatedObjects]).to.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.not.be.undefined;
+
+        expect(model[$materials][5][$correlatedObjects]).to.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.not.be.undefined;
+
+        await model[$switchVariant]('Peacock Velvet');
+
+        expect(model[$materials][4][$correlatedObjects]).to.not.be.null;
+        expect(model[$materials][4][$lazyLoadGLTFInfo]).to.be.undefined;
+
+        expect(model[$materials][5][$correlatedObjects]).to.not.be.null;
+        expect(model[$materials][5][$lazyLoadGLTFInfo]).to.be.undefined;
+      });
+
+      test(
+          'Switch back to default variant does not change correlations',
+          async () => {
+            const threeGLTF = await loadThreeGLTF(SHEEN_CHAIR_GLB_PATH);
+            const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+
+            const sizeBeforeSwitch =
+                model[$materials][0][$correlatedObjects]!.size;
+
+            await model[$switchVariant]('Peacock Velvet');
+            // Switches back to default.
+            await model[$switchVariant]('Mango Velvet');
+
+            expect(model[$materials][0][$correlatedObjects]!.size)
+                .equals(sizeBeforeSwitch);
+          });
+
+      test('Switching variant when model has no variants', async () => {
+        const threeGLTF = await loadThreeGLTF(KHRONOS_TRIANGLE_GLB_PATH);
+        const model = new Model(CorrelatedSceneGraph.from(threeGLTF));
+
+        const threeMaterial =
+            model[$materials][0][$correlatedObjects]!.values().next().value;
+        const sizeBeforeSwitch = model[$materials][0][$correlatedObjects]!.size;
+        await model[$switchVariant]('Does not exist');
+
+        expect(model[$materials][0][$correlatedObjects]!.values().next().value)
+            .equals(threeMaterial);
+        expect(model[$materials][0][$correlatedObjects]!.size)
+            .equals(sizeBeforeSwitch);
+      });
+    });
```

Comment:
Might be nice to add a test that you can look up all the material names without causing them to load. Also that you can access an unloaded material without crashing (since that should cause it to load).

Reply:
Done.